### PR TITLE
Add NativeAOT build and runtime pack for linux-bionic

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
Build and runtime pack for linux-bionic (Android without the Java part).

I tried not to regress the existing CoreCLR Android build that is based on a crossrootfs (#56622) - the if's basically deal with that. Note that it's likely broken anyway (#66562).

Cc @dotnet/ilc-contrib

### CODE REVIEW

**Comment by steveisok** [comment]
@MichalStrehovsky can you kick off an official build of this branch to make sure it's ok?

**Comment by MichalStrehovsky** [comment]
> @MichalStrehovsky can you kick off an official build of this branch to make sure it's ok?

Triggered this build: https://dnceng.visualstudio.com/internal/_build/results?buildId=2188707&view=results

I _think_ it looks good? It failed at signing, but I assume it's because I triggered it and it's not the main branch.

File: eng/native/build-commons.sh [hunk 1]
```diff
@@ -84,6 +84,8 @@ build_native()
             exit 1
         fi
 
+        cmakeArgs="-C $__RepoRootDir/eng/native/tryrun.cmake $cmakeArgs"
```
Comment (akoeplinger) [inline eng/native/build-commons.sh:87]: why is this necessary?

**Review by akoeplinger — APPROVED** [review]
One question but looks good to me otherwise.

**Comment by akoeplinger** [comment]
> Triggered this build: [dnceng.visualstudio.com/internal/_build/results?buildId=2188707&view=results](https://dnceng.visualstudio.com/internal/_build/results?buildId=2188707&view=results)
>
> I think it looks good? It failed at signing, but I assume it's because I triggered it and it's not the main branch.

It should work for non-main builds as well but I checked and I've seen the same error on some 8.0-preview5 branch builds as well so I think it is unrelated.

File: eng/native/build-commons.sh (hunk 1 above)
Comment (akoeplinger) [reply eng/native/build-commons.sh:87]: Yes, what I meant was why is this not necessary for the mono-based Android build today?

I looked a bit deeper now and I think it is because we're not using any of these cmake checks in the System.Native build so we don't hit this, e.g. we explicitly skip the checks that would cause binaries to be run during configure on mobile and manually set the results: 
https://github.com/dotnet/runtime/blob/c5256458235e00c4bc4634c3cc389fefdd4c10d8/src/native/libs/configure.cmake#L572-L605

So this is fine since we'll override the values for e.g. `HAVE_SHM_OPEN_THAT_WORKS_WELL_ENOUGH_WITH_MMAP` but we should probably clean this up and move all of the detection into tryrun.cmake. Would you mind filing an issue for that?

Comment (MichalStrehovsky) [reply eng/native/build-commons.sh:87]: Filed #86829.
